flocker/node/_deploy.py

```diff
@@ -733,7 +793,13 @@ def change_node_state(deployer, desired_configuration,  current_cluster_state):
 
     :return: ``Deferred`` that fires when the necessary changes are done.
     """
-    d = deployer.discover_local_state()
+    nodes = [node for node in current_cluster_state.nodes
+             if node.hostname == deployer.hostname]
+    if nodes:
+        node = [nodes[0]]
```

Comment:
I suspect an acceptance test will catch this. Stupid legacy flocker-changestate stuff.